modules/key-distributor/main.tf

```
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cfn_files = [
+    "${path.module}/cfn/encrypter/function.go",
+    "${path.module}/cfn/encrypter/go.mod",
+  ]
+  cfn_md5sums     = [for f in local.cfn_files : filemd5(f)]
+  cfn_dirchecksum = md5(join("-", local.cfn_md5sums))
+}
+
+resource "google_storage_bucket" "cfn_bucket" {
+  project       = var.project_id
+  name          = "${var.project_id}-cfn-${var.function_name}"
+  location      = "US"
+  force_destroy = true
+}
+
+data "archive_file" "cfn" {
```

Let's use `function` instead of `cfn` throughout (since it's not an intuitive acronym).

```suggestion
data "archive_file" "function" {
```